src/index.ts

```diff
@@ -0,0 +1,88 @@
+import crypto from 'crypto';
+import { Request, RequestHandler, NextFunction } from 'express';
+
+import stringToBuffer from './stringToBuffer';
+import validateArguments from './validateArguments';
+import AuthError from './errors';
+
+export type Options = {
+    algorithm: string;
+    identifier: string;
+    header: string;
+    maxInterval: number;
+    minInterval: number;
+}
+
+const defaults: Options = {
+    algorithm: 'sha256',
+    identifier: 'HMAC',
+    header: 'authentication',
```

Comment:
Not really familiar with this type of auth but Is this default correct? Should it be `authorization` instead?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0b5829c771a62066831c806748857c1cab1b69ea/types/node/http.d.ts#L23

Reply:
I think you may be right, even though HMAC is a form of authentication rather than authorization - the [ITEF spec does state you can use digest based authentication in the `authorization` header](https://datatracker.ietf.org/doc/html/rfc2617#section-3.3).

Funnily enough AWS themselves state [the header is poorly named](https://docs.aws.amazon.com/AmazonS3/latest/userguide/RESTAuthentication.html#ConstructingTheAuthenticationHeader) in their documentation on their implementation of HMAC.

Reading the rest of the spec this package should support the WWW-Authentication header, I will add that to the backlog.